api/app/Providers/AuthServiceProvider.php

```diff
@@ -92,8 +92,12 @@ public function resolveUserOrAbort($bearerToken, $tokenService): ?User
 
             // By this point we have verified that the token is legitimate
 
-            $userMatch = User::where('sub', $sub)->first(); // 3. match "sub" claim to user 'sub' field.
+            $userMatch = User::where('sub', $sub)->withTrashed()->first(); // 3. match "sub" claim to user 'sub' field.
             if ($userMatch) {
+                if ($userMatch->deleted_at != null) {
```

Comment:
Hmm, I'm worried about this.  Even though the app pages are guarded the deleted user can still interact with the API like before.  Could you return a 401 with an error message that is recognized by the app?

Reply:
Updated to prevent calls from deleted users with [27fce5b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7593/commits/27fce5b8526556960ec66f77a6439dd478cb0a97).

Reply:
Nice!